rust/xaynet-server/src/storage/impls.rs

```diff
@@ -334,3 +334,34 @@ pub enum SumDictDeleteError {
     #[error("sum participant does not exist")]
     DoesNotExist,
 }
+
+#[derive(Deref)]
+pub struct MaskDictIncr(Result<(), MaskDictIncrError>);
+
+impl MaskDictIncr {
+    pub fn into_inner(self) -> Result<(), MaskDictIncrError> {
+        self.0
+    }
+}
+
+impl FromRedisValue for MaskDictIncr {
+    fn from_redis_value(v: &Value) -> RedisResult<MaskDictIncr> {
+        match *v {
+            Value::Int(-2) => Ok(MaskDictIncr(Err(MaskDictIncrError::MaskAlreadySubmitted))),
+            Value::Int(-1) => Ok(MaskDictIncr(Err(MaskDictIncrError::UnknownSumPk))),
+            Value::Int(0) => Ok(MaskDictIncr(Ok(()))),
```

Comment:
thought that occurred to me: it would be nice to be able to map "error codes" to error variants right at the point of definition, by making the error codes discriminants of the enum variants i.e.
```rust
pub enum MaskDictIncrError {
    UnknownSumPk = -1,
    MaskAlreadySubmitted = -2,
}
```
but it seems it's not straightforward to convert from an int to an enum, there's at least [some boilerplate](https://stackoverflow.com/a/57578431) required. so I'm not sure whether it's worth it. but might be worth considering.

Reply:
Not a bad idea🤔 Maybe we can use `#[derive(TryInto)]` to hide the boilerplate code. I will try it out and give an update on the result.